pyface/ui/qt4/util/gui_test_assistant.py

```diff
@@ -148,6 +148,31 @@ def event_loop_until_condition(self, condition, timeout=10.0):
         except ConditionTimeoutError:
             self.fail('Timed out waiting for condition')
 
+    def assertEventuallyTrueInGui(self, condition, timeout=10.0):
+        """
+        Assert that the given condition becomes true if we run the GUI
+        event loop for long enough.
+
+        This assertion runs the real Qt event loop, polling the condition
+        and returning as soon as the condition becomes true. If the condition
+        does not become true within the given timeout, the assertion fails.
+
+        Parameters
+        ----------
+        condition : callable() -> bool
+            Callable accepting no arguments and returning a bool.
+        timeout : float
+            Maximum length of time to wait for the condition to become
+            true, in seconds.
+
+        Raises
+        ------
+        self.failureException
+            If the condition does not become true within the given timeout.
+        """
+        with self.event_loop_until_condition(condition, timeout=timeout):
```

Comment:
This might be better if we just used the event loop helper's `event_loop_until_condition` inside an appropriate try/except.  This implementation adds cognitive load to developers who come after and are trying to understand the implementation.

Reply:
Agreed; I'll fix it.

Reply:
Done.